flow: faster, leaner single openroad invocation flows

oharboe opened the pull request:
no longer save out all .odb/.sdc files automatically, the user can save out as many or as few as they need.

oharboe committed:
flow: faster, leaner single openroad invocation flows

no longer save out all .odb/.sdc files automatically,
the user can save out as many or as few as they need.

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

oharboe committed:
scripts: introduce orfs_write_db/sdc wrappers

This commit is only to simplify the review.

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

maliberty commented:
> no longer save out all .odb/.sdc files automatically

I don't see a change in the default flow.  I suppose you mean it in the context of the SINGLE_OPENROAD_INVOCATION_FLOW.   

SINGLE_OPENROAD_INVOCATION_FLOW doesn't actually control the invocation of OR so it feels odd.  Further you might want a single invocation flow to write out an .odb at each step even if it doesn't exit, so this feels orthogonal.

oharboe committed:
openroad: single invocation flow has more knobs

erasing variables or not and saving sdc/odb after each stage are now different
variables

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

maliberty commented on flow/scripts/floorplan_to_place.tcl:
```diff
@@ -1,5 +1,6 @@
 # Enable keeping variables between stages
 set ::env(KEEP_VARS) 1
+set ::env(WRITE_ODB_AND_SDC_EACH_STAGE) 0
```
This will prevent the user from overriding it.  I think this should only be set if currently unset.

maliberty replied:
The intent of the script is to perform all operations in a single invocation.  Whether to write odb at each stage or not is a separate decision.  @eder-matheus does this matter for you?

eder-matheus replied:
In the scenario I'm using it, I'm interested on the ODB files of each stage. However, I'm not against having it as an option, and the default behavior would be not writing the ODBs of each stage.

oharboe committed:
floorplan_to_place:  behavior is back to what it was on master

This PR now adds the option of skipping saving of
.odb/.sdc files only

Signed-off-by: Øyvind Harboe <oyvind.harboe@zylin.com>

oharboe commented:
@eder-matheus @maliberty Should be all good now. Anything else?
